lib/app.go

```diff
@@ -184,11 +183,11 @@ func appStateInMutablePod(app *v1.App, pod *pkgPod.Pod) error {
 	finishedAt := fi.ModTime().UnixNano()
 	app.FinishedAt = &finishedAt
 
-	// Read exit code.
-	exitCode, err := readExitCode(appStatusFile)
-	if err != nil {
-		return err
-	}
+	// if we can't get the exit code (e.g. we didn't have time to write it
+	// after a pod dies) let's continue instead of erroring out, the exit code
+	// will be -1 and we'll let callers handle it.
+	code, _ := pod.AppExitCode(app.Name)
+	exitCode := int32(code)
```

Comment:
Exit codes should be in the 0-255 range though (u8). I've no idea if this would confuse any consumer, but it looks a bit sketchy. I suggest to handle the error instead and fallback to either `1` (generic error code) or `254` (rkt error code).

Reply:
That sounds reasonable. Thanks!

Reply:
I went with a generic error code since we don't know what really happened.